vowpalwabbit/OjaNewton.cc

```diff
@@ -52,51 +52,53 @@ struct OjaNewton {
     bool normalize;
     bool random_init;
 
-    void initialize_Z()
-    {
-        size_t stride_shift = all->reg.stride_shift;
-        weight* weights = all->reg.weight_vector;
-
-        uint32_t length = 1 << all->num_bits;
-        if(normalize) { // initialize normalization part
-	    for (uint32_t i = 0; i < length; i++)
-	        weights[(i << stride_shift) + NORM2] = 0.1;
-        }
-
+	void initialize_Z()
+	{  weight_vector& weights = *(all->wv);
+       uint32_t length = 1 << all->num_bits;
+	   if (normalize) { // initialize normalization part
+		 for (weight_vector::iterator i = weights.begin(NORM2); i != weights.end(NORM2); ++i)
+		   *i = 0.1;
```

Comment:
I'm afraid of this operation because it will blow up with the number of parameters as I understand it. Maybe it's better to have a set_default() interface which for dense parameters does this but for the hashmap operates in a lazy fashion?